pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -609,6 +616,35 @@ func (r *Store) getDynamicField(data map[string]interface{}) (string, error) {
 	return "", nil
 }
 
+func (r *Store) updateClusterByK8sclient(id, name string, data map[string]interface{}) (map[string]interface{}, error) {
+	logrus.Tracef("Updating cluster [%s] using K8s dynamic client", id)
+	if r.ClusterClient == nil {
+		return nil, fmt.Errorf("Error updating the cluster: k8s client is nil")
+	}
+
+	object, err := r.ClusterClient.Get(context.TODO(), id, metav1.GetOptions{})
```

Comment:
Pass in the apiContext here and use it vs doing a TODO
```suggestion
	object, err := r.ClusterClient.Get(apiContext.Request.Context(), id, metav1.GetOptions{})
```

Reply:
And in the other spot TODO is used

Reply:
Updated